subsys/net/l2/wifi/wifi_mgmt.c

```diff
@@ -1583,21 +1551,12 @@ static int __stored_creds_to_params(struct wifi_credentials_personal *creds,
 #ifdef CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ENTERPRISE
 	if (params->security == WIFI_SECURITY_TYPE_EAP_TLS) {
 		if (creds->header.key_passwd_length > 0) {
-			key_passwd = (char *)k_malloc(creds->header.key_passwd_length + 1);
-			if (!key_passwd) {
-				LOG_ERR("Failed to allocate memory for key_passwd\n");
-				ret = -ENOMEM;
-				goto err_out;
-			}
-			memset(key_passwd, 0, creds->header.key_passwd_length + 1);
-			ret = snprintf(key_passwd, creds->header.key_passwd_length + 1, "%s",
-				       creds->header.key_passwd);
-			if (ret > creds->header.key_passwd_length) {
+			if (creds->header.key_passwd_length > WIFI_ENT_PSWD_MAX_LEN) {
 				LOG_ERR("key_passwd string truncated\n");
```

Comment:
Not your change but the `\n` should not be there as it adds extra newline to log output. Could you fix these as I guess there are more than one of these in this file, in a separate commit?